namesys/dns.go

```diff
@@ -50,24 +55,46 @@ func (r *DNSResolver) resolveOnce(ctx context.Context, name string) (path.Path,
 	if !isd.IsDomain(segments[0]) {
 		return "", errors.New("not a valid domain name")
 	}
-
 	log.Infof("DNSResolver resolving %s", segments[0])
-	txt, err := r.lookupTXT(segments[0])
+
+	prefixes := []string{"", "_dnslink."}
+
+	resChan := make(chan lookupRes)
+
+	for _, prefix := range prefixes {
+		go workDomain(r, prefix + segments[0], resChan)
```

Comment:
Point is that it would be best done concurrently but wit `_dnslink.` starting first. This way on Unix they will be fully concurrent and in case of everything else sequential.

If someone throws different hashes into both places, sorry but in my opinion it is his problem. Doing it sequentially creates performance trade-off for someone saying conflicting things (which will be really rare).